app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,47 @@
+class OrdersController < ApplicationController
+  before_action :load_categories, only: %i(index)
+
+  def index
+    return redirect_to root_path if current_user.nil?
+    
+    @pagy, @orders = pagy(load_orders, items: Settings.order.paginates)
+  end
+
+  def create
+    @order = current_user.orders.build(order_params)
+    cart_items = JSON.parse(cookies[:cart_items])
+    @order.build_order_items_from_cart cart_items
+
+    if @order.save
+      cookies.delete(:cart_items)
+
+      flash[:success] = "Order successfully placed!"
+      redirect_to orders_path
+    else
+      flash[:error] = "Order failed!"
+    end
+  end
+
+  def update
+    redirect_to orders_path, notice: "Order was successfully canceled!" if cancel_order Order.find_by(params[:id])
```

Comment:
vẫn câu hỏi cũ, ko tìm thấy order thì như thế nào em

Reply:
em fix rồi nha anh, khi sd find_by nếu không tìm thấy nó trả ra cái đầu tiên nên biến order vẫn có giá trị, e check id của nó có đúng không nếu sai thì xuống dưới là Unable to cancel order